src/odemis/gui/model/tab_gui_data.py

```diff
@@ -346,6 +398,41 @@ def __init__(self, main):
 
         # Current tool selected (from the toolbar)
         tools = {TOOL_NONE, TOOL_RULER, TOOL_FEATURE}
+        if main.stigmator:
+            stig = getComponent(role="stigmator")
```

Comment:
Don't crate a new connection to the HwComponent, just do `stig = main.stigmator`, or just `stig_md = main.stigmator.getMetadata()`.